src/ConnectionFromMongoCursor.js

```diff
@@ -170,7 +170,10 @@ const connectionFromMongoCursor = async ({
   cursor.skip(skip);
   cursor.limit(limit);
 
-  const slice = await cursor.exec();
+  //avoid large object retrieval from database
+  const clonedCursor = cursor.find().merge(cursor).select({ _id: 1 })
```

Comment:
I don't know which you prefer. IMHO a js function needs to be pure, and calling .skip, .limit, .select, etc. on a mongoose Query alter the object itself.

Let's assume:

```
const users = UserModel.find(where, { _id: 1 }).sort({
    createdAt: -1,
  });

  return connectionFromMongoCursor({
    cursor: users,
    context,
    args,
    loader: load,
  });
```

If I call


```
await users
```

after the invocation of connectionFromMongoCursor, the result is unexpected , because it's filtered (and projected if you use cursor.find().select({_id: 1}) inside connectionFromMongoCursor).

Maybe the best choice is to use 

```
const clonedCursor = cursor.find().merge(cursor)
```

and always use clonedCursor inside connectionFromMongoCursor.